### DISCUSSION TITLE
Telegram calls only rings but not get connected

### DISCUSSION THREAD

**Discussion by mostafaznv** [v2fly/v2ray-core#2082] [Q&A]
## What version of V2Ray are you using?

[v5.1.0 User Preview](https://github.com/v2fly/v2ray-core/releases/tag/v5.1.0)

## What's your scenario of using V2Ray?

Using Telegram **calls** with Telegram Mac application via internal socks5 proxy (vmess)

## What problems have you encountered?

Telegram call rings, but not get connected and I can't see any UDP connection in my console log.

## What's your expectation?

I want to make phone calls with telegram on my macbook.
I'm using `v2ray cli` on my mac and `shadowrocket` on my iPhone. 
With iPhone, I can make calls without any problems, but I can't do that using macbook.

 
![Screen Shot 1401-07-25 at 5 43 31 PM](https://user-images.githubusercontent.com/7619687/196200161-ab888b49-9b7a-4c15-8e77-64b52017f7c0.png)

## Please attach your configuration here

**Server configuration:**

```javascript
{
  "inbounds": [
    {
      "port": 1000,
      "listen": "127.0.0.1",
      "protocol": "vmess",
      "settings": {
        "udp": true,
        "clients": [
          {
            "id": "uuid",
            "alterId": 0
          }
        ]
      },
      "streamSettings": {
        "network": "ws",
        "wsSettings": {
          "path": "/path"
        }
      }
    }
  ],
  "outbounds": [
    {
      "protocol": "freedom",
      "settings": {}
    }
  ]
}
```

**Client configuration:**

```javascript
{
  "inbounds": [
    {
      "listen": "127.0.0.1",
      "port": 1082,
      "protocol": "socks",
      "sniffing": {
        "enabled": true,
        "destOverride": ["http", "tls"]
      },
      "settings": {
        "auth": "noauth",
        "udp": true
      }
    }
  ],
  "outbounds": [
    {
      "protocol": "vmess",
      "settings": {
        "vnext": [
          {
            "address": "host",
            "port": 443,
            "users": [
              {
                "id": "uuid",
                "alterId": 0
              }
            ]
          }
        ]
      },
      "streamSettings": {
        "network": "ws",
        "security": "tls",
        "wsSettings": {
          "path": "/path"
        }
      }
    }
  ]
}
```

## Please attach error logs here

There is no error logs

## Please attach access log here

There is no error logs

## Other configurations (such as Nginx) and logs here

```
server {
    server_name host;

    location /path {
        if ($http_upgrade != "websocket") {
            return 404;
        }

        proxy_redirect off;
        proxy_pass http://127.0.0.1:1000;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_set_header Host $host;

        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    }

    listen 443 ssl;
    ssl_certificate /path/fullchain.pem;
    ssl_certificate_key /path/privkey.pem;
    include /etc/letsencrypt/options-ssl-nginx.conf;
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; 
}

```

**Comment by kulongwangzhi85** [comment]
download geoip https://github.com/Loyalsoldier/v2ray-rules-dat
"routing": {
  ...
  {
          "type": "field",
          "outboundTag": "xxx-tag",
          "network": "udp,tcp",
          "ip": [
	    "8.8.8.8",
	    "8.8.4.4",
	    "1.1.1.1",
	    "1.0.0.1",
	    "9.9.9.9",
	    "geoip:cloudflare",
	    "geoip:cloudfront",
	    "geoip:facebook",
	    "geoip:fastly",
	    "geoip:netflix",
	    "geoip:twitter",
	    "geoip:telegram",
	    "geoip:google"
          ]
        },
  ... ...
}

**Reply by mostafaznv** [reply]
Nothing changed.

**Note**: I'm using telegram through sock5 proxy (127.0.0.1:1082).
